### ISSUE TITLE
dubbo2.7.8升级到dubbo3.0后,无法启动

### ISSUE THREAD

**Issue by iwin1983** [apache/dubbo#8199]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.0
* Operating System version: windows10
* Java version: 1.8
* Registry : zookeeper 3.4.12
* Spring Boot : 2.1.3.RELEASE

### Steps to reproduce this issue

dubbo2.7.8升级到dubbo3.0后, Provider无法启动

### Expected Result

What do you expected from the above steps？

### Actual Result

What actually happens?

If there is an exception, please attach the exception trace:

```
java.lang.NullPointerException: null
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.isAnnotatedReferenceBean(ReferenceAnnotationBeanPostProcessor.java:165)
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.postProcessBeanFactory(ReferenceAnnotationBeanPostProcessor.java:131)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:286)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:174)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:705)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:531)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:142)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:775)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:316)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
2021-07-01 21:49:50 logback [DubboShutdownHook] ERROR o.a.d.c.bootstrap.DubboBootstrap -  [DUBBO] Dubbo config was cleaned prematurely, dubbo version: 3.0.0, current host: 172.16.1.68
Exception in thread "DubboShutdownHook" java.lang.RuntimeException: java.lang.IllegalStateException: Dubbo config was cleaned prematurely
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:48)
	at org.apache.dubbo.common.lang.ShutdownHookCallbacks.lambda$callback$0(ShutdownHookCallbacks.java:70)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.apache.dubbo.common.lang.ShutdownHookCallbacks.callback(ShutdownHookCallbacks.java:70)
	at org.apache.dubbo.config.DubboShutdownHook.callback(DubboShutdownHook.java:77)
	at org.apache.dubbo.config.DubboShutdownHook.run(DubboShutdownHook.java:64)
Caused by: java.lang.IllegalStateException: Dubbo config was cleaned prematurely
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.checkConfigState(DubboBootstrap.java:1481)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.destroy(DubboBootstrap.java:1448)
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:46)
	... 5 more
```

**Comment by codergmc** [comment]
Can you provide a minimal project that we can reproduce the issue?

**Comment by iwin1983** [comment]
of course，how to send to you?

**Comment by codergmc** [comment]
share on github

**Comment by iwin1983** [comment]
https://github.com/iwin1983/demo

**Comment by codergmc** [comment]
Please upgrade to 3.0.1.
Reference [3.0.1 ](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1  ) 

> Fix NPE when consumer start (#8080)

**Comment by iwin1983** [comment]
It's ok in Dubbo 3.0.1, Thank you for your help!
